### PR TITLE
[perf] reduce radix cache match overhead by changing the match algorithm

### PR DESCRIPTION
**Pull request by Qiaolin-Yu**
## Accuracy Tests
Before,
image

After,
image

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #27039543572](https://github.com/sgl-project/sglang/actions/runs/27039543572)
Latest PR Test (Extra): :white_check_mark: [Run #27044672508](https://github.com/sgl-project/sglang/actions/runs/27044672508)

### CODE REVIEW

File: python/sglang/srt/mem_cache/radix_cache.py [hunk 1]
```diff
@@ -135,37 +135,39 @@ def _check_compatible(self, other: "RadixKey") -> None:
...
             )
 
-    # TODO(Jialin): replace zip with numpy to skip per-element PyLong boxing
     def match(self, other: "RadixKey", page_size: int = 1) -> int:
         """Logical-unit prefix length shared with ``other``. Result is rounded down to ``page_size``."""
         self._check_compatible(other)
         t0, t1 = self.token_ids, other.token_ids
+        n = min(len(t0), len(t1))
+
+        # Exponential search for the first diverging token: gallop in doubling
```
Comment (ispobock) [inline python/sglang/srt/mem_cache/radix_cache.py:145]: Add ut to cover edge case and page size > 1

**Review by hzh0425 — APPROVED** [review]
Looks good！

Maybe need to add some unit tests

**Comment by vladnosiv** [comment]
Hi, I recently did a similar PR, except for this place on long contexts up to a million tokens, there is another limitation with RadixKey slicing: https://github.com/sgl-project/sglang/pull/26943

**Comment by hzh0425** [comment]
> Hi, I recently did a similar PR, except for this place on long contexts up to a million tokens, there is another limitation with RadixKey slicing: #26943

will forward you pr after this pr merged

File: python/sglang/srt/mem_cache/radix_cache.py (hunk 1 above)
Comment (Qiaolin-Yu) [reply python/sglang/srt/mem_cache/radix_cache.py:145]: done
